File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/ImplicitEqualsSelector.java
package org.evomaster.client.java.controller.mongo.selectors;

import org.evomaster.client.java.controller.mongo.operations.*;

import static org.evomaster.client.java.controller.mongo.utils.BsonHelper.getValue;
import static org.evomaster.client.java.controller.mongo.utils.BsonHelper.documentKeys;

import java.util.Map;
import java.util.Set;

/**
 * { field: value }
 */
public class ImplicitEqualsSelector extends QuerySelector {
    @Override
    public QueryOperation getOperation(Object query) {
        String fieldName = extractFieldName(query);
        if (!isUniqueEntry((Map<?, ?>) query)) return null;
        Object value = getValue(query, fieldName);
        return new EqualsOperation<>(fieldName, value);
    }

    protected String extractOperator(Object query) {
        String fieldName = extractFieldName(query);
        Set<String> keys = documentKeys(getValue(query, fieldName));
        return keys.stream().findFirst().orElse(null);
    }

    @Override
    protected String operator() {
        return "";
    }

    private String extractFieldName(Object query) {
        Set<String> keys = documentKeys(query);
        return keys.stream().findFirst().orElse(null);
    }
}

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/ImplicitEqualsSelector.kt
package org.evomaster.client.java.controller.mongo.selectors

import org.bson.Document
import org.evomaster.client.java.controller.mongo.operations.*

/**
 * { field: value }
 */
class ImplicitEqualsSelector : QuerySelector() {
    override fun getOperation(query: Document): QueryOperation? {
        val fieldName = extractFieldName(query)
        if (!isUniqueEntry(query)) return null
        val value = query[fieldName]
        return EqualsOperation(fieldName, value)
    }

    override fun extractOperator(query: Document): String {
        val fieldName = extractFieldName(query)
        return  (query[fieldName] as Document).keys.first()
    }

    override fun operator(): String {
        return ""
    }

    private fun extractFieldName(query: Document): String = query.keys.first()
}